Production_scripts/convert_gp_costs.R

```diff
@@ -0,0 +1,143 @@
+#####################################################
+# Costs - GP Out of Hours
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description -
+#####################################################
+
+
+## update file ##
+
+## Attendances taken from 2018 Primary Care Out of Hours Report ##
+# https://publichealthscotland.scot/publications/out-of-hours-primary-care-services-in-scotland/
+
+## Costs taken from R520 (Costbook) report for 2015/16 ##
+# https://beta.isdscotland.org/topics/finance/costs/ (R520)
+
+## The above should be checked / added to the Excel file 'OOH_Costs.xlsx' before running this syntax.
+
+
+library(dplyr)
+library(ggplot2)
+
+
+## data ##
+gp_out_of_hours <- readxl::read_xlsx(paste0(get_slf_dir(), "/Costs/OOH_Costs.xlsx"))
+
+
+## data - wide to long ##
+gp_out_of_hours <-
+  gp_out_of_hours %>%
+  pivot_longer(
+    contains("Consultations"),
+    names_to = "Year_Consultations", values_to = "Consultations"
+  ) %>%
+  pivot_longer(
+    contains("Cost"),
+    names_to = "Year_Cost", values_to = "Cost"
+  )
+
+
+## create year variable ##
+gp_out_of_hours <-
+  gp_out_of_hours %>%
+  mutate(Year = substr(Year_Consultations, 1, 4))
+
```

Comment:
We can use some nice features of `pivot_longer` to make this much simpler / shorter.
```suggestion
  gp_out_of_hours %>%
  pivot_longer(c(ends_with("_Consultations"), ends_with("_Cost")),
    names_to = c("year", ".value"),
    names_pattern = "(\\d{4})_(.+)"
  )
```